torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1167,6 +1160,12 @@ std::vector<std::shared_ptr<NCCLComm>>& ProcessGroupNCCL::getNCCLComm(
   // [Note 2 ]
   C10D_NCCL_CHECK(ncclGroupEnd(), c10::nullopt);
 
+  // At this point NCCL should have been initialized, hence we can accurately get
+  // the env value even if NCCL sets it by reading from nccl.conf file
+  if (getRank() == 0) {
+    LOG(INFO) << "NCCL_DEBUG: " << parse_env("NCCL_DEBUG");
+  }
```

Comment:
Shouldn't we print this on all ranks? The log above "ProcessGroupNCCL initialized with following options" is printed on all ranks.

Reply:
It is more for verbosity consideration - not sure if a user wants 1000 lines of `NCCL_DEBUG=INFO` printed in the log, for something not that critical. The other options, by contrast, need to be consistent across all ranks, otherwise there would be a hang. So there may still a value in printing them on all ranks.